### PR TITLE
feat(bindings): add `useStateManager()`

### PR DESCRIPTION
**Pull request by layershifter**
### `@stardust-ui/state`

- `createManager()` - functionality to create state manager, based on @levithomason prototype (only async handling was removed)
- `createDialogManager()` & `createDropdownManager()` - basic managers that prove that it works

### `@stardust-ui/react-bindings`

- `useStateManager()` - React hook that connects state managers with props and implements auto controlled pattern

***

#### 📡 POC

https://codesandbox.io/s/dropdown-prototype-r3ux9
(_prototype of `Dropdown` component_)

### CODE REVIEW

File: packages/react-bindings/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "@stardust-ui/react-bindings",
+  "description": "TODO",
```
Comment (layershifter) [inline packages/react-bindings/package.json]: Add description

File: packages/react-bindings/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "react-is": "^16.6.3"
+  },
+  "devDependencies": {
+    "@stardust-ui/internal-tooling": "^0.38.0",
+    "lerna-alias": "^3.0.3-0"
+  },
+  "files": [
+    "dist"
+  ],
+  "homepage": "https://github.com/stardust-ui/react/tree/master/packages/react-component-ref",
```
Comment (layershifter) [inline packages/react-bindings/package.json]: Fix homepage

File: packages/state/.babelrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "@stardust-ui/internal-tooling/babel"
+}
```
Comment (layershifter) [inline packages/state/.babelrc]: Cruft file

File: packages/state/package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@stardust-ui/state",
+  "description": "TODO",
```
Comment (layershifter) [inline packages/state/package.json]: Add description

File: packages/state/package.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  "author": "Oleksandr Fediashov <a@fedyashov.com>",
+  "bugs": "https://github.com/stardust-ui/react/issues",
+  "devDependencies": {
+    "@stardust-ui/internal-tooling": "^0.38.0",
+    "lerna-alias": "^3.0.3-0"
+  },
+  "files": [
+    "dist"
+  ],
+  "homepage": "https://github.com/stardust-ui/react/tree/master/packages/react-component-ref",
```
Comment (layershifter) [inline packages/state/package.json]: Fix homepage

File: packages/state/src/createManager.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      applyMiddleware()
+      applySideEffects()
+    }
+
+    manager.actions[actionName] = enhancedAction
+  })
+
+  const applyAction = (action: Action<State, ActionNames>, ...args: any[]) => {
+    if (!action) return
+    if (debug) console.log('manager ACTION', action.name || 'Anonymous')
```
Comment (layershifter) [inline packages/state/src/createManager.ts]: Strip in production

File: packages/state/src/createManager.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+    const actionResult = action(...args)(getState(), actions)
+    if (actionResult) setState(actionResult)
+  }
+
+  const applyMiddleware = () => {
+    const prevState = getState()
+
+    middleware.forEach((middlewareItem, index) => {
+      if (debug) {
```
Comment (layershifter) [inline packages/state/src/createManager.ts]: Strip in production

File: packages/state/src/managers/dialogManager.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import createManager from '../createManager'
+import { Manager, ManagerConfig } from '../types'
+
+export type DialogActionNames = 'open' | 'close'
```
Comment (layershifter) [inline packages/state/src/managers/dialogManager.ts]: Definition should allow to define params for actions

**Review by jurokapsiar — APPROVED** [review]

**Pull request merged**